packages/eas-cli/src/credentials/ios/actions/new/RemoveProvisioningProfile.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  AppleProvisioningProfileFragment,
+  AppleProvisioningProfileIdentifiersFragment,
+} from '../../../../graphql/generated';
+import log from '../../../../log';
+import { confirmAsync } from '../../../../prompts';
+import { Action, CredentialsManager } from '../../../CredentialsManager';
+import { Context } from '../../../context';
+import { AppLookupParams } from '../../api/GraphqlClient';
+
+/* interface RemoveOptions {
+  shouldRevoke?: boolean;
+}
+
+export class RemoveProvisioningProfile implements Action {
+  constructor(private accountName: string, private options: RemoveOptions = {}) {}
+
+  async runAsync(manager: CredentialsManager, ctx: Context): Promise<void> {
+    const selected = await selectProvisioningProfileFromExpoAsync(ctx, this.accountName);
+    if (selected) {
+      const app = getAppLookupParams(selected.experienceName, selected.bundleIdentifier);
+      await manager.runActionAsync(new RemoveSpecificProvisioningProfile(app, this.options));
+    }
+  }
+} */
+
+interface RemoveSpecificOptions {
+  shouldRevoke?: boolean;
+}
+
+export class RemoveSpecificProvisioningProfile implements Action {
+  constructor(
+    private app: AppLookupParams,
+    private provisioningProfile: AppleProvisioningProfileIdentifiersFragment,
+    private options: RemoveSpecificOptions = {}
+  ) {}
+
+  async runAsync(manager: CredentialsManager, ctx: Context): Promise<void> {
+    await ctx.newIos.deleteProvisioningProfileAsync(this.provisioningProfile.id);
+    log.succeed(
+      `Removed provisioning profile for @${this.app.account.name}/${this.app.projectName} (${this.app.bundleIdentifier})`
+    );
+
+    const developerPortalIdentifier = this.provisioningProfile.developerPortalIdentifier;
+    // We need the developer portal identifier to revoke the profile on Apple
+    if (!developerPortalIdentifier) {
+      return;
+    }
+
```

Comment:
I'm not sure if we need to revoke the profile on the apple developer portal,
- there can be only one profile for bundle id (for specific type)
- we already support reusing/updating provisioning profile when "creating" a new one

those calls could take some time when someone has revoked dist cert used in 10 apps and it does not change anything from the user perspective, during the next build user still will need to generate new profile.

Reply:
yeah this makes sense. i removed the revoking logic for Apple developer portal, exposed `deleteProvisioningProfiles` on www, so we only make one request to delete a bunch of profiles